packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -256,11 +256,10 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
       // Create a mapping of chunks to dependent chunks, used to add the proper
       // link tags for CSS.
       for (const chunk of pureCSSChunks) {
-        const referenceId = chunkToReferenceIdMap.get(chunk.fileName)!;
-        const chunkReferenceIds = [referenceId];
-        for (const imp of chunk.imports) {
-          if (chunkToReferenceIdMap.has(imp)) {
-            chunkReferenceIds.push(chunkToReferenceIdMap.get(imp)!);
+        const chunkReferenceIds: string[] = [];
+        for (const [specifier, chunkRefID] of chunkToReferenceIdMap.entries()) {
```

Comment:
This is the main change: `chunkToReferenceMap` contains the correct import order. When we use that, we more-or-less get styles in the order imported.

`chunk.imports` seems to be more-or-less in alphabetical order. So iterating over that will cause styles to be rearranged on build.

Reply:
Ah, good catch.